src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1081,19 +1081,22 @@ function ($a) {
                         ->arrayNode('buses')
                             ->defaultValue(array('messenger.bus.default' => array('default_middleware' => true, 'middleware' => array())))
                             ->useAttributeAsKey('name')
-                            ->prototype('array')
+                            ->arrayPrototype()
                                 ->addDefaultsIfNotSet()
                                 ->children()
-                                    ->booleanNode('default_middleware')->defaultTrue()->end()
+                                    ->enumNode('default_middleware')
+                                        ->values(array(true, false, 'allow_no_handlers'))
+                                        ->defaultTrue()
+                                    ->end()
                                     ->arrayNode('middleware')
                                         ->beforeNormalization()
-                                            ->ifString()
-                                            ->then(function (string $middleware) {
-                                                return array($middleware);
+                                            ->always()
+                                            ->then(function ($middleware) {
+                                                return \is_string($middleware) || !\is_int(key($middleware)) ? array($middleware) : $middleware;
```

Comment:
Should we put `\is_string($middleware) || !\is_int(key($middleware))` inside `ifTrue()` instead of using `always()`? Would be consistent with the rest of the nodes I guess.

Reply:
makes sense, updated